internal/migrations/observability/api/output_test.go

```diff
@@ -0,0 +1,974 @@
+package api
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	openshiftv1 "github.com/openshift/api/config/v1"
+
+	logging "github.com/openshift/cluster-logging-operator/api/logging/v1"
+	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/gcl"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("#ConvertOutputs", func() {
+	Context("default output", func() {
+		It("should generate default elasticsearch output based on logstoreSpec", func() {
+			logStoreSpec := logging.LogStoreSpec{
+				Type: logging.LogStoreTypeElasticsearch,
+			}
+			expEsOut := &obs.OutputSpec{
+				Name: "default-elasticsearch",
+				Type: obs.OutputTypeElasticsearch,
+				Elasticsearch: &obs.Elasticsearch{
+					URLSpec: obs.URLSpec{
+						URL: fmt.Sprintf("https://%s:%d", string(obs.OutputTypeElasticsearch), 9200),
+					},
+					Version: 6,
+					IndexSpec: obs.IndexSpec{
+						Index: "{{.log_type}}",
+					},
+				},
+				TLS: &obs.OutputTLSSpec{
+					TLSSpec: obs.TLSSpec{
+						CA: &obs.ConfigMapOrSecretKey{
+							Secret: &corev1.LocalObjectReference{
+								Name: constants.CollectorName,
+							},
+							Key: constants.TrustedCABundleKey,
+						},
+						Certificate: &obs.ConfigMapOrSecretKey{
+							Secret: &corev1.LocalObjectReference{
+								Name: constants.CollectorName,
+							},
+							Key: constants.ClientCertKey,
+						},
+						Key: &obs.SecretKey{
+							Secret: &corev1.LocalObjectReference{
+								Name: constants.CollectorName,
+							},
+							Key: constants.ClientPrivateKey,
+						},
+					},
+				},
+			}
+			Expect(generateDefaultOutput(&logStoreSpec)).To(Equal(expEsOut))
+		})
+
+		It("should generate default lokistack output", func() {
```

Comment:
should this still produce 3 outputs?

Reply:
No this is converting `default` lokistack to an explicit lokistack output. During migration of obs.Clf, this will get converted into the appropriate outputs/pipelines based on inputs.